src/v/http/tests/request_builder_test.cc

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "http/request_builder.h"
+
+#include <gtest/gtest.h>
+
+namespace bh = boost::beast::http;
+
+TEST(request_builder, test_default_build_is_error) {
+    http::request_builder rb;
+    auto req = rb.build();
+    ASSERT_TRUE(!req.has_value());
+    ASSERT_TRUE(req.error() == rb.default_state);
+}
+
+TEST(request_builder, test_with_everything) {
+    http::request_builder rb;
+    auto req = rb.host("https://foobar.xyz")
+                 .path("san_mig")
+                 .with_bearer_auth("secret")
+                 .with_content_type("obsidian")
+                 .with_content_length(1)
+                 .header("one", "1")
+                 .headers({{"two", "2"}, {"three", "3"}})
+                 .method(boost::beast::http::verb::delete_)
+                 .query_param("here")
+                 .query_param_kv("name", "no")
+                 .query_params({"g", "b", "u"})
+                 .query_params_kv({{"trilogy", "of"}, {"containers", "four"}})
+                 .build();
+    ASSERT_TRUE(req.has_value());
+    const auto& built = req.value();
+    using enum bh::field;
+    ASSERT_EQ(built[host], "foobar.xyz");
+    ASSERT_EQ(built[authorization], "Bearer secret");
+    ASSERT_EQ(built[content_length], "1");
+    ASSERT_EQ(built[content_type], "obsidian");
+    ASSERT_EQ(built["one"], "1");
+    ASSERT_EQ(built["two"], "2");
+    ASSERT_EQ(built["three"], "3");
+    ASSERT_TRUE(built.method() == bh::verb::delete_);
```

Comment:
not `EXPECT_EQ`?

Reply:
I think I got a compilation error at some point and misread it as the verb enum class not implementing `operator<<` but it seems to compile fine now. changed to `EXPECT_EQ`